users/src/main/java/io/crate/operation/auth/HostBasedAuthentication.java

```diff
@@ -163,12 +170,12 @@ static boolean isValidProtocol(String hbaProtocol, HbaProtocol protocol) {
             return hbaProtocol == null || hbaProtocol.equals(protocol.toString());
         }
 
-        static boolean isValidConnection(String hbaConnectionMode, HbaProtocol protocol) {
+        static boolean isValidConnection(String hbaConnectionMode, ConnectionProperties connectionProperties) {
             return hbaConnectionMode == null ||
                    hbaConnectionMode.isEmpty() ||
                    hbaConnectionMode.equals(SSL_OPTIONS.OPTIONAL.VALUE) ||
-                   (hbaConnectionMode.equals(SSL_OPTIONS.NEVER.VALUE) && protocol == HbaProtocol.POSTGRES) ||
-                   (hbaConnectionMode.equals(SSL_OPTIONS.REQUIRED.VALUE) && protocol == HbaProtocol.POSTGRES_SSL);
+                   (hbaConnectionMode.equals(SSL_OPTIONS.NEVER.VALUE) && !connectionProperties.hasSSL()) ||
+                   (hbaConnectionMode.equals(SSL_OPTIONS.REQUIRED.VALUE) && connectionProperties.hasSSL());
```

Comment:
👍 I had the same idea in my PR where I introduced `hasSsl()` on the `HbaProtocol` enum. Much clearer.